framework/src/main/java/org/tron/common/logsfilter/capsule/TransactionLogTriggerCapsule.java

```diff
@@ -156,6 +163,100 @@ public TransactionLogTriggerCapsule(TransactionCapsule trxCapsule, BlockCapsule
                     StringUtil.encode58Check(createSmartContract.getOwnerAddress().toByteArray()));
               }
               break;
+            case UnfreezeBalanceContract:
+              UnfreezeBalanceContract unfreezeBalanceContract = contractParameter
+                  .unpack(UnfreezeBalanceContract.class);
+
+              if (Objects.nonNull(unfreezeBalanceContract)) {
+                if (Objects.nonNull(unfreezeBalanceContract.getOwnerAddress())) {
+                  transactionLogTrigger.setFromAddress(StringUtil
+                      .encode58Check(unfreezeBalanceContract.getOwnerAddress().toByteArray()));
+                }
+                if (Objects.nonNull(unfreezeBalanceContract.getReceiverAddress())) {
```

Comment:
The frequency of these if statements is too high, you can encapsulate a function.

Reply:
OK, Ok, I encapsulate the code